src/libs/actions/Report.ts

```diff
@@ -2696,17 +2696,44 @@ function updateGroupChatMemberRoles(reportID: string, accountIDList: number[], r
         participants[accountID] = {role};
     });
 
+    const optimisticParticipants: Participants = {};
+    const successParticipants: Participants = {};
+
+    Object.keys(participants).forEach((accountID) => {
+        const participantID = Number(accountID);
+        optimisticParticipants[participantID] = {
+            ...participants[participantID],
+            pendingFields: {
+                role: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            },
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+        };
```

Comment:
We shouldn't need a `participants` variable and we can avoid the extra loop and use the first one.

```js
    const memberRoles: Record<number, string> = {};
    const optimisticParticipants: Participants = {};
    const successParticipants: Participants = {};

    accountIDList.forEach((accountID) => {
        memberRoles[accountID] = role;
        optimisticParticipants[accountID] = {
            role,
            pendingFields: {
                role: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
            },
            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
        };
        successParticipants[accountID] = {
            pendingFields: {
                role: null,
            },
            pendingAction: null,
        };
    });
```